src/components/mortgage-investment-compare/input-card.js

```diff
@@ -0,0 +1,286 @@
+import React from 'react'
+
+import Accordion from './accordion'
+import {
+  Card,
+  Col,
+  Form,
+  FormGroup,
+  Input,
+  InputGroup,
+  InputGroupAddon,
+  InputGroupText,
+  Row
+} from 'reactstrap'
+
+import { formatMoney } from '../../utils/numberUtils'
+
+const InputCard = props => {
+  return (
+    <Card className='p-4'>
+      <Form>
+        <FormGroup>
+          <label>How much is your loan?</label>
+          <InputGroup>
+            <InputGroupAddon addonType='prepend'>
+              <InputGroupText>$</InputGroupText>
+            </InputGroupAddon>
+            <Input
+              name='loanAmt'
+              placeholder='Loan Amount'
+              type='number'
+              value={props.loanAmt}
```

Comment:
(see above) I'm a bit concerned that we are modifying props internally. This is similar to if a child function modifies the arguments, b/c it's now stateful

Reply:
see comment above

Reply:
Maybe an agnostic example would show what I mean by stateful:

```js
function getName (person) {
  person.lastName = _getLastName(person.name)
  return person.name
}

person = {name: 'Sally Seashell'}
console.log(getName(person)) // 'Sally'
console.log(person) // {name: 'Sally', lastName: 'Seashell'} Uh oh
```

Here `props` is person, and when someone types something, we modify `props` directly. This breaks an expectation that `props` is immutable

Reply:
we are not modifying props directly in the child. It runs through `updateInput`, which runs `setState`. We really haven't changed the way the state is being modified.

Reply:
`setState` reads from `this.state`, which is what we pass into `InputCard` as `props`, right?

Reply:
yes but it does it within the parent, so it never talks to props. It talks to the original `this.state`. This is essentially what we're doing: https://ourcodeworld.com/articles/read/409/how-to-update-parent-state-from-child-component-in-react

It's very common

Reply:
https://stackoverflow.com/questions/35537229/how-to-update-parents-state-in-react

Reply:
👍 In your examples, the child is calling a function, which the parent handles. It doesn't modify data directly, so it matches the pattern I was expecting earlier

Reply:
@hdavidzhu yes that is what we are doing here correct? Am I missing something?

Reply:
Yep, what's happening is that we are modifying `this.state` of the parent directly, since `state === props`

Instead, in the examples, they call a handler (much like triggering an event, like `onInputChange(data)`, which then the parent handles by doing `onInputChange: (data) -> this.setState('something', data)`

Reply:
I'll make this non-blocking though, b/c the feature works! Children directing modifying arguments could lead to some future gotchas, but we can cross that bridge later then

Reply:
No, we are calling a handler too just like in the examples @hdavidzhu. It's `this.updateInput`. The state will only be modified through `this.updateInput` which only runs in the parent. Can you point at the exact place where `props` is being modified directly?

Reply:
> Children directing modifying arguments could lead to some future gotchas

'props' is never being modified directly by the `Child`. I would correct this immediately if I saw it, and we would already see issues with the calculations if this was happening. Please show me exactly where you are seeing this.

Reply:
This is the process flow:

1. Input is modified by the user (loan amount in this case)
2. `updateInput` (the handler) gets called and runs in the parent
3. `updateInput` changes the state of the parent with `this.setState()`
4. Since the state changes, new props are passed into the `Child` component, reflecting the changes in the `Parent`'s state 

![](https://giphy.com/gifs/Qx5Fzb2MjYSXtdypxs)

Reply:
> since `state === props`

Also this is incorrect. `props` are just a set of arguments being passed into the `Child`. Some of these arguments live in the `Parent`'s state, others live within the `render()` etc. But that doesn't really matter as props is never modified directly.

Reply:
Ah yep, you are absolutely correct. We read the data from the event. Thanks!

Reply:
Yes. You'll see me doing this a lot in react. I just wanted to make sure you knew exactly what was happening to avoid confusion later on. 

I don't remember ever having to modify `props` directly but I'm pretty sure that if we were to do that, things would just not work (at least in our app) because modifying `props` doesn't modify the parent's `state`, and every child pulls from the parent's `state` right now. The results on the right wouldn't change if `props` were being modified directly.

Please feel free to resolve when you see this comment.